src/main/java/rx/Observable.java

```diff
@@ -4547,6 +4547,398 @@ public final void onNext(T args) {
     }
 
     /**
+     * Concatenates up to 2 sources eagerly into a single stream of values.
+     * 
+     * <p>
+     * Eager concatenation means that once a Subscriber subscribes, all sources are subscribed to, their values fully buffered and
+     * they are drained in order, one after the previous completes.
+     * 
+     * <dl>
+     *  <dt><b>Backpressure:</b></dt>
+     *  <dd>Backpressure is honored towards the downstream, however, due to the eagerness requirement, sources are subscribed to
+     *  in unbounded mode and values queued up in an unbounded buffer.</dd>
+     *  <dt><b>Scheduler:</b></dt>
+     *  <dd>This method does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     * @param <T> the value type 
+     * @param o1
+     * @param o2
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> Observable<T> eagerConcatMap(Observable<? extends T> o1, Observable<? extends T> o2) {
+        return eagerConcatMap(Arrays.asList(o1, o2));
+    }
+    
+    /**
+     * Concatenates up to 3 sources eagerly into a single stream of values.
+     * 
+     * <p>
+     * Eager concatenation means that once a Subscriber subscribes, all sources are subscribed to, their values fully buffered and
+     * they are drained in order, one after the previous completes.
+     * 
+     * <dl>
+     *  <dt><b>Backpressure:</b></dt>
+     *  <dd>Backpressure is honored towards the downstream, however, due to the eagerness requirement, sources are subscribed to
+     *  in unbounded mode and values queued up in an unbounded buffer.</dd>
+     *  <dt><b>Scheduler:</b></dt>
+     *  <dd>This method does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     * @param <T> the value type
+     * @param o1
+     * @param o2
+     * @param o3
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> Observable<T> eagerConcatMap(
```

Comment:
Should we name these `eagerConcat` instead of `eagerConcatMap`? These overloads do not take a `FuncN` so they aren't really mapping.